clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -53,11 +53,21 @@ private object ApiClient {
   }
 }
 
-class ApiClient(apiUrl: String, accessKey: String, secretKey: String) {
+class ApiClient(
+    apiUrl: String,
+    accessKey: String,
+    secretKey: String,
+    connectionTimeout: String = ""
+) {
   private val client = new api.ApiClient
   client.setUsername(accessKey)
   client.setPassword(secretKey)
   client.setBasePath(apiUrl.stripSuffix("/"))
+  if (connectionTimeout != null && !connectionTimeout.isEmpty) {
+    // converting the connection timeout from seconds to milliseconds
+    val connectionTimeoutMillisec = connectionTimeout.toInt * 1000
```

Comment:
please use scala constant for milliseconds 
import scala.concurrent.duration.{Duration, MILLISECONDS}, see this for example https://www.programcreek.com/scala/scala.concurrent.duration.MILLISECONDS